internal/provider/file_mode.go

```diff
@@ -0,0 +1,61 @@
+package provider
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"strconv"
+
+	"github.com/hashicorp/go-cty/cty"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+const defaultFileMode = "0444"
+
+func convertStrToFileMode(s string) (os.FileMode, error) {
+	if s == "" {
+		s = defaultFileMode
+	}
+	a, err := strconv.ParseInt(s, 8, 32)
+	if err != nil {
+		return 0, err
+	}
+	if a < 0 {
+		return 0, fmt.Errorf("file_mode must be greater equal than 0: %d", a)
+	}
+	if a > 0o777 {
+		return 0, fmt.Errorf("file_mode must be less equal than 0o777: %d", a)
```

Comment:
Limiting it to `0o777` means limits the mode to just `[ugo][rwx]` -- it should also be possible to set the setuid, setgid, and sticky bits.  To allow these, use `0o4777` as the maximum value.

Also, does `ParseInt(s, 8, 32)` handle the situation when the string is leads with `0` and `0o`?

Do we also want to keep the current behavior of assuming the string is a _decimal_ value, unless the string leads with a `0` or `0o`?  If not, we should make it very clear in the changelog (etc.) that the workaround will no longer work.